Python/TelemetrySystem/telemetry.py

```diff
@@ -1,25 +1,25 @@
-from client import TelemetryClient
+from client import AbstractClient
+
 
 class TelemetryDiagnostics:
-    DiagnosticChannelConnectionString = "*111#"
+    DIAGNOSTIC_CHANNEL_CONNECTION_STRING = "*111#"
 
-    def __init__(self):
-        self._telemetry_client = TelemetryClient()
+    def __init__(self, client: AbstractClient, retries: int = 3):
+        self.retries = retries
+        self._telemetry_client = client
         self.diagnostic_info = ""
 
-    def check_transmission(self):
+    def check_transmission(self, client_message: str):
         self.diagnostic_info = ""
         self._telemetry_client.disconnect()
 
-        retryLeft = 3
-        while ((not self._telemetry_client.online_status) and retryLeft > 0):
-            self._telemetry_client.connect(TelemetryDiagnostics.DiagnosticChannelConnectionString)
-            retryLeft -= 1
+        retry_left = self.retries
+        while not self._telemetry_client.online_status and retry_left > 0:
+            self._telemetry_client.connect(self.DIAGNOSTIC_CHANNEL_CONNECTION_STRING)
+            retry_left -= 1
```

Comment:
Looks like the logic relates only to the connection itself, not to Diagnostics itself, doesn't it?

Reply:
Do you mean to create a separate method to connect to the client before we run check_transmission ?

Reply:
Nope, it means that you just need to call .connect and it should take care of retrying if it's necessary.